### PR TITLE
added wishlist api 

### PR DESCRIPTION
**Pull request by JatuFaique**
- minor fixes in product listing page 
- handled authorization errors and scenarios

### CODE REVIEW

File: src/components/ProductCard.js [hunk 1]
```diff
@@ -44,18 +44,61 @@ function ProductCard({ data, isWishlist }) {
...
-      payload: prod,
-    });
+  const handleAddtoWishlist = async (prod) => {
+    try {
+      wishlistDispatch({
+        type: "REQUEST_ADD_TO_WISHLIST",
+      });
+
+      const encT = localStorage.getItem("token");
+      console.log(encT, prod);
```
Comment (prathmeshjagtap) [inline src/components/ProductCard.js:54]: you can remove console.log

File: src/components/ProductCard.js [hunk 2]
```diff
@@ -44,18 +44,61 @@ function ProductCard({ data, isWishlist }) {
...
 
-  const handleRemoveWishlist = (prod) => {
-    wishlistDispatch({
-      type: "REMOVE_FROM_WISHLIST",
-      payload: prod,
-    });
+  const handleRemoveWishlist = async (prod) => {
+    try {
+      const encT = localStorage.getItem("token");
+      console.log(encT, prod);
```
Comment (prathmeshjagtap) [inline src/components/ProductCard.js:83]: same here

File: src/components/ProductCard.js [hunk 3]
```diff
@@ -44,18 +44,61 @@ function ProductCard({ data, isWishlist }) {
...
+      const res = await axios.delete(`/api/user/wishlist/${prod._id}`, {
+        headers: {
+          authorization: encT,
+        },
+      });
+
+      if (res.status == 200 || res.status == 201) {
+        console.log("resp = ", res.data);
```
Comment (prathmeshjagtap) [inline src/components/ProductCard.js:91]: +1

**Review by prathmeshjagtap — APPROVED** [review]
Rest is good

File: src/reducers/cartReducer.js [hunk 4]
```diff
@@ -37,43 +37,6 @@ function cartReducer(state, action) {
...
-    //     return {
-    //       ...state,
-    //       cart: t_arr,
-    //     };
-    //   } else {
-    //     console.log(t_arr);
-    //     return {
-    //       ...state,
-    //     };
-    //   }
```
Comment (mcmonojit) [inline src/reducers/cartReducer.js:76]: If you don't plan to use this code then make sure to remove it.

File: src/styles/styles.css [hunk 5]
```diff
@@ -298,6 +299,12 @@
   background-color: #ececec;
```
Comment (mcmonojit) [inline src/styles/styles.css:299]: use var() for colours

File: src/styles/styles.css [hunk 6]
```diff
@@ -262,6 +262,7 @@
   border: 1px solid #cccccca8;
```
Comment (mcmonojit) [inline src/styles/styles.css:262]: use var() for colours

**Review by mcmonojit — APPROVED** [review]
Looks good to merge.

**Pull request merged**